lib/clients/ZookeeperManager.js

```diff
@@ -0,0 +1,313 @@
+const zookeeper = require('node-zookeeper-client');
+const EventEmitter = require('events');
+const async = require('async');
+
+class ZookeeperManager extends EventEmitter {
+    /**
+     * Constructs an instance of the ZookeeperManager.
+     *
+     * @param {string} connectionString - The connection string for the ZooKeeper server.
+     * @param {Object} options - Configuration options for the ZooKeeper client.
+     * @param {Logger} log - An instance of a logging utility.
+     */
+    constructor(connectionString, options, log) {
+        super();
+
+        this.connectionString = connectionString;
+        this.log = log;
+        this.options = options;
+        this.client = null;
+
+        this._connect();
+    }
+
+    /**
+     * Establishes a connection to the ZooKeeper server.
+     *
+     * This method initializes a new ZooKeeper client using the provided connection string
+     * and options. It sets up event listeners for various client states such as 'connected',
+     * 'disconnected', and 'expired'. The method handles reconnection logic in case of
+     * client expiration and forwards ZooKeeper events through this class's EventEmitter interface.
+     *
+     * If 'autoCreateNamespace' is set in the options and the base path does not exist in ZooKeeper,
+     * this method tries to create the necessary path (namespace) in ZooKeeper.
+     *
+     * @method _connect
+     * @return {undefined}
+     */
+    _connect() {
+        // clean up exists client before reconnect
+        if (this.client) {
+            this.client.removeAllListeners();
+        }
+
+        this.client = zookeeper.createClient(this.connectionString, this.options);
```

Comment:
Structurally I think we should turn the whole construct into an async series or waterfall to avoid nesting too many levels. Also we would then be able to treat success/error in a unique place at the end of the series and emit the appropriate `ready` or `error` event.